File: web/vroom/src/app/core/services/registered-user.service.ts
import { HttpClient, HttpParams } from "@angular/common/http";
import { Inject, Injectable } from "@angular/core";
import { RideResponseDTO } from "../models/ride/responses/ride-respose.dto";
import { Observable } from "rxjs";

@Injectable({
    providedIn: 'root'
})
export class RegisteredUserService{
    private api = "http://localhost:8080/api/registered-user"

    constructor(private http: HttpClient){}

    getRideHistoryRequest(
        pageNum: number,
        pageSize: number,
        startDate?: any,
        endDate?: any,
        sortBy?: 'startTime,asc' | 'startTime,desc' | 'price,asc' | 'price,desc'
    ): Observable<RideResponseDTO[]>{
        let params = new HttpParams()
            .set('sort', sortBy || 'startTime,desc')
            .set('pageNumber', pageNum.toString())
            .set('pageSize', pageSize.toString())
            
        if (startDate) {
            const start = new Date(startDate);
            params = params.set('startDate', start.toISOString());
        }
        if (endDate) {
            const end = new Date(endDate);
            params = params.set('endDate', end.toISOString());
        }
        return this.http.get<RideResponseDTO[]>(`${this.api}/rides`, { params })
    } 
}

File: mobile/app/src/main/java/com/example/vroom/services/RegisteredUserService.java
package com.example.vroom.services;

import com.example.vroom.DTOs.ride.responses.RideResponseDTO;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RegisteredUserService {
    @GET("api/registered-user/rides")
    Call<List<RideResponseDTO>> getRides(
            @Query("sort") String sort,
            @Query("startDate") String startDate,
            @Query("endDate") String endDate,
            @Query("pageNumber") int pageNumber,
            @Query("pageSize") int pageSize
    );
}
